### ISSUE TITLE
Issue with New-HPOVServerProfileTemplate and firmware baseline?

### ISSUE THREAD

**Issue by tknightblc** [HewlettPackard/POSH-HPOneView#195]
Chris,

Using latest OV 3.0 POSH library, cannot seem to get a New-HPOVServerProfileTemplate command to accept a firmware baseline. I had this working in OV 2.0 with following format:

```PowerShell
New-HPOVServerProfileTemplate -name "A BL460 Gen8 ESX Template No SAN" -description "Template for ESX host-BL460 Gen8 with no SAN storage definitions" -baseline "Service Pack For ProLiant" -FirmwareMode FirmwareOffline -serverHardwareType $bl460Gen8SHT -enclosureGroup $eg -connections $conListA –manageBoot -pxeBootPolicy Auto -bios -biossettings $Gen8BiosSettings  @params   | Wait-HPOVTaskComplete
```

The SPP name is still generic "Service Pack For ProLiant" but it does not assign - and I put gibberish in it ignores as well. I also tried this syntax:

```PowerShell
$bline = Get-HPOVBaseline -file "871790_001_spp-2016_10_0-SPP2016100_2016_1015_191.iso" -ApplianceConnection 192.168.136.100

New-HPOVServerProfileTemplate -name "test1" -description "Template for ESX host-BL660 Gen9 with no SAN storage definitions"  -baseline $bline -FirmwareMode FirmwareOffline -serverHardwareType $bl660Gen9SHT -enclosureGroup $eg -connections $conListA –manageBoot -pxeBootPolicy Auto -bios -biossettings $Gen9BiosSettings -Bootmode UEFI  @params   | Wait-HPOVTaskComplete
```

And still ignores - no errors, just does not assign baseline. 

Did something change in OV 3.00 for this? Again it worked on OV 2.00.

I can provide more detail if needed. Version info below.

Tom

```PowerShell
PowerShell Library Version : 3.0.1170.1773
Library Path               : C:\Program Files\WindowsPowerShell\Modules\HPOneView.300




Name                           Value                                                                                 
----                           -----                                                                                 
PSVersion                      4.0                                                                                   
WSManStackVersion              3.0                                                                                   
SerializationVersion           1.1.0.1                                                                               
CLRVersion                     4.0.30319.42000                                                                       
BuildVersion                   6.3.9600.16406                                                                        
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0}                                                                  
PSRemotingProtocolVersion      2.2                                                                                   
```

**Comment by ChrisLynchHPE** [comment]
First off, you are missing the `-Firmware` parameter switch.  Without it, the firmware parameters you have been providing will be ignored (which is how it has always worked): 

```PowerShell

New-HPOVServerProfileTemplate -name "test1" -description "Template for ESX host-BL660 Gen9 with no SAN storage definitions" -Firmware -baseline $bline -FirmwareMode FirmwareOffline -serverHardwareType $bl660Gen9SHT -enclosureGroup $eg -connections $conListA –manageBoot -pxeBootPolicy Auto -bios -biossettings $Gen9BiosSettings -Bootmode UEFI  @params | Wait-HPOVTaskComplete
```

Second, please use a Baseline Object, and not the name as there is a slight defect with how Baseline resource Name is handled with the current version.

**Comment by tknightblc** [comment]
Chris,

Thanks. Embarrassed I missed the –firmware flag, I had that in my OV 2.0 scripts not sure how I removed it.

I will give it a try today.

Happy New Year!

                                                                Thanks, Tom K.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 3.0.1191.2858](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v3.0.1191.2858).

Closing.
